rewrite-pattern.js

```diff
@@ -704,9 +775,13 @@ const rewritePattern = (pattern, flags, options) => {
 		transform(options, 'unicodeFlag') || transform(options, 'unicodePropertyEscapes')
 	);
 	config.transform.namedGroups = transform(options, 'namedGroups');
+	config.transform.modifiers = transform(options, 'modifiers');
+
+	config.modifiersData = Object.create(null);
```

Comment:
Can we just use `Object.create(null)` above when we create the `modifiersData` object? Or better, can we use a known shape? `modifiersData: { s: false, m: false, ... }`.

Reply:
This is a global variable and we need to reset it on every call because I use it to determine whether the feature is enabled or not.
I made some changes.